(GH-36) Restructure documentation

gep13 opened the pull request:
Based heavily on what Cake.Issues.Recipe has done.  Thanks @pascalberger

Relates to #36

AdmiringWorm commented on docs/input/docs/contributing/how-to-build.md:
````diff
@@ -0,0 +1,18 @@
+
+```bash
+./build.sh
+```
````
I would recommend having a note mentioning not to run the `setup.cake` script directly, as we do some additional changes in the bootstrappers to make the dogfooding work as expected.

AdmiringWorm commented on docs/input/docs/index.cshtml:
```diff
@@ -1,11 +1,10 @@
-<p>We need your help to improve the documentation for Cake.Recipe, so if there is something that you 
-would like to add then you can edit the content directly on GitHub.</p>
+<p>
+    We gladly accept your contributions. If there is something that you would like to add then you can edit the content directly on GitHub.
```
Only partially true, they can edit on GitHub but they would need to open a PR to have any updates being available on the documentation site.
Maybe linking to `how to contribute` documentation as well?

AdmiringWorm commented on docs/input/docs/contributing/how-to-contribute.md:
```diff
@@ -0,0 +1,19 @@
+* Push your feature branch.
+* Create a pull request.
+
+For getting started see issues marked with `Up-for-grabs` in the individual repositories.
```
We only have 1 repository for Cake.Recipe, and we don't really use the `Up for grabs` label.
I haven't seen it used at least.

gep13 commented:
@AdmiringWorm I believe I have addressed your comments here as well.   Would be good to get this merged in, so that we can continue making small PR's to update docs, rather than one massive PR getting out of hand.
